walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,67 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+    // Find times that don't work
+    ArrayList<TimeRange> badRanges = new ArrayList<>();
+    for (Event event : events) {
+      Collection<String> eventAttendees = event.getAttendees();
+      for (String attendee : attendees) {
+        if (eventAttendees.contains(attendee)) {
+          badRanges.add(event.getWhen());
+          break;
+        }
+      }
+    }
+    badRanges.sort(TimeRange.ORDER_BY_START);
+
+    // Prune time ranges
+    for (int i = 0; i < badRanges.size() - 1; i++) {
+      TimeRange range1 = badRanges.get(i);
+      TimeRange range2 = badRanges.get(i + 1);
+      // Range2 can contain range1 if they have equal start times, but that case is equivalent to
+      // overlap
+      if (range1.overlaps(range2)) {
+        if (range1.contains(range2)) {
+          // Irrelevant time range
+          badRanges.set(i + 1, null);
+        } else {
+          // Merge overlapping
+          TimeRange combined = TimeRange.fromStartEnd(range1.start(), range2.end(), false);
```

Comment:
if 3 consecutive ranges i, i+1 and i+2 should all be merged, it is not clear that you are doing this.